File: app/src/main/java/com/example/myapplication/view/auth/LoginFragment.java
package com.example.myapplication.view.auth;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;

import com.example.data.api.AuthApi;
import com.example.data.dto.LoginRequest;
import com.example.myapplication.R;
import com.example.myapplication.databinding.FragmentLoginBinding;

import javax.inject.Inject;

import dagger.hilt.android.AndroidEntryPoint;
import retrofit2.Call;
import retrofit2.Response;

@AndroidEntryPoint
public class LoginFragment extends Fragment {

    FragmentLoginBinding binding;

    @Inject
    AuthApi authApi; //Retrofit을 사용하여 서버에 인증 관련 요청 보내는 데 사용

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentLoginBinding.inflate(inflater, container, false);

        binding.login.setOnClickListener(v -> {
            LoginRequest loginRequest = new LoginRequest(binding.emailEditText.getText().toString(), binding.passwordEditText.getText().toString());
            Log.i("LoginFragment", loginRequest.toString());
            getActivity().runOnUiThread(() -> {
                Toast.makeText(getActivity(), "로그인 시도중", Toast.LENGTH_SHORT).show();

                try {
                    Call<String> call = authApi.login(loginRequest);
                    call.enqueue(new retrofit2.Callback<String>() {
                        @Override
                        public void onResponse(@NonNull Call<String> call, @NonNull Response<String> response) {
                            if (response.isSuccessful()) {
                                String jwtToken = response.body();
                                if (jwtToken != null) {
                                    Toast.makeText(getActivity(), "로그인 성공", Toast.LENGTH_SHORT).show();
                                    Log.d("LoginFragment", jwtToken.toString());
                                }
                            }
                        }

                        @Override
                        public void onFailure(@NonNull Call<String> call, @NonNull Throwable t) {
                            Toast.makeText(getActivity(), "로그인 실패", Toast.LENGTH_SHORT).show();
                            Log.e("LoginFragment", t.getMessage());
                        }
                    });
                } catch (Exception e) {
                    Log.e("LoginFragment", e.getMessage());
                }
            });
        });

        binding.signup.setOnClickListener(v -> {
            Navigation.findNavController(v).navigate(R.id.action_loginFragment2_to_signUpInputEmailFragment2);
        });

        binding.emailFind.setOnClickListener(v -> {
            Navigation.findNavController(v).navigate(R.id.action_loginFragment2_to_emailFindFragment);
        });

        return binding.getRoot();
    }

}

File: app/src/main/java/com/example/myapplication/view/auth/LoginFragment.kt
package com.example.myapplication.view.auth

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.lifecycleScope
import androidx.navigation.Navigation
import com.example.data.api.AuthApi
import com.example.data.dto.LoginRequest
import com.example.data.dto.TokenResponse
import com.example.myapplication.common.MySharedPreferences
import com.example.myapplication.databinding.FragmentLoginBinding
import com.example.myapplication.view.home.HomeActivity
import com.example.myapplication.viewmodel.AuthViewModel
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.launch
import retrofit2.Response
import javax.inject.Inject

@AndroidEntryPoint
class LoginFragment : Fragment() {

    @Inject
    lateinit var authApi: AuthApi

    private lateinit var binding: FragmentLoginBinding
    private lateinit var authViewModel: AuthViewModel

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentLoginBinding.inflate(inflater, container, false)
        authViewModel = ViewModelProvider(requireActivity())[AuthViewModel::class.java]
        binding.viewModel = authViewModel

        binding.loginButton.setOnClickListener {
            lifecycleScope.launch {
                val loginRequest: LoginRequest = authViewModel.getLoginRequest()
                val response: Response<TokenResponse> = authApi.login(loginRequest)
                if (response.isSuccessful) {
                    val token: String = response.body()!!.token!!
                    Toast.makeText(context, "로그인 성공 토큰: $token", Toast.LENGTH_SHORT).show()
                    MySharedPreferences.setUserInfo(
                        requireContext(),
                        loginRequest.email,
                        loginRequest.password
                    )
                    MySharedPreferences.setToken(requireContext(), token)

                    // 로그인 성공 시 EventActivity를 시작
                    val intent = Intent(requireContext(), HomeActivity::class.java)
                    startActivity(intent)
                    requireActivity().finish()
                } else {
                    Toast.makeText(context, "로그인 실패", Toast.LENGTH_SHORT).show()
                }
            }
        }

        binding.signup.setOnClickListener {
            val action = LoginFragmentDirections.actionLoginFragment2ToSignUpInputEmailFragment2()
            Navigation.findNavController(binding.root).navigate(action)
        }

        binding.emailFind.setOnClickListener {
            val action = LoginFragmentDirections.actionLoginFragment2ToEmailFindFragment()
            Navigation.findNavController(binding.root).navigate(action)
        }

        return binding.root
    }
}
